src/main/org/deidentifier/arx/ARXConfiguration.java

```diff
@@ -707,7 +707,9 @@ public Monotonicity getMonotonicityOfPrivacy() {
      * @return
      */
     public Monotonicity getMonotonicityOfUtility() {
-        if (metric.isMonotonic() || (this.getMaxOutliers() == 0d) || this.isPracticalMonotonicity()) {
+        if ((metric.isMonotonicWithGeneralization() && this.getMaxOutliers() == 0) || 
```

Comment:
Consider this a false alarm.  I went back through my local history this morning, tested out my early code that didn't tie generalization to outliers, and tested a purposeful mismatch of outliers and monotonicity types.  Unit tests caught both cases.  I think I was already worried about results and convinced myself something was amiss.